12.js

```diff
@@ -0,0 +1,31 @@
+const stringMeAlong = string => {
+  let longestSubstring = ''
+  let substring
+
+  for (let index = 0; index < string.length; index++) {
```

Comment:
You could avoid searching for more substrings when the `longestSubstring` you have now is longer than the amount of string you have left to check because it couldnt not possibly contain a longer substring.